docs/example-usage.ipynb

```diff
@@ -0,0 +1,1478 @@
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "5beaa9a4-e219-4ef5-b25c-796c59db9c5a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[ 0. 75.  0.  0.]\n",
+      " [90. 15.  0.  0.]\n",
+      " [60.  0. 25.  0.]\n",
+      " [ 0.  0. 20. 10.]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(grid)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a4a96a5c-33b6-4ab5-a653-8c55a920fe7a",
+   "metadata": {},
+   "source": [
+    "\n",
+    "This is the essentially how Level 2 Gridded data is inflated and what is happening in the service.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58be0de9-3007-4ba5-9bbc-d0860c2374bc",
+   "metadata": {},
+   "source": [
+    "## Using the service\n",
+    "\n",
+    "use [harmony-py](https://github.com/nasa/harmony-py) and initialize a UAT Harmony client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "6b0ca429-899a-4ece-9592-120fac5b04e1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from harmony import Client, Collection, Environment, Request\n",
+    "\n",
+    "harmony_client = Client(env=Environment.UAT)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c8a81d3-03e8-4300-83f7-af099c0aacf5",
+   "metadata": {},
+   "source": [
+    "Create a Harmony Request using a known SPL2SMA granule for testing.\n",
+    "\n",
+    "The request has:\n",
+    "- a collection for SPL2SMA,\n",
+    "- a granule_id that crosses a large section of land\n",
+    "- a crs of WGS84 and allows harmony to know to invoke this service \n",
+    "- a format to say we want netcdf output.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "31ec80ec-a8c5-42f7-aaf9-cfa8bfba29d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_request = Request(\n",
+    "    collection=Collection(id='C1268429309-EEDTEST'),\n",
+    "    granule_id='G1268454418-EEDTEST',\n",
+    "    crs='EPSG:4326',\n",
+    "    format='application/x-netcdf4',\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f565e043-6c60-4e2d-87a6-dd7b96b09cc7",
+   "metadata": {},
+   "source": [
+    "Submit the request to harmony and download the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "3ec88bf4-0996-4c7d-b374-b605c35e3b1d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "5209028_SMAP_L2_SM_P_00872_D_20150401T044811_R19240_001_regridded.nc\n",
+      "Downloaded processed file from Harmony: 5209028_SMAP_L2_SM_P_00872_D_20150401T044811_R19240_001_regridded.nc\n"
+     ]
+    }
+   ],
+   "source": [
+    "job_id = harmony_client.submit(test_request)\n",
+    "\n",
+    "for filename in [\n",
+    "    file_future.result()\n",
+    "    for file_future in harmony_client.download_all(job_id, overwrite=True)\n",
```

Comment:
Probably just an opinion...

I think what you are doing here is the most efficient in terms of lines of code (as `harmony.Client.download_all` will wait for processing to complete), but I think it might be a bit more user friendly if you used the `harmony.Client.wait_for_processing` function before, so you could use `show_progress=True`. It's mainly an aesthetic thing, but I think it's clearer/nicer for a user to see something saying Harmony is working on the request.